packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts

```diff
@@ -507,11 +569,38 @@ export class PluginLedgerConnectorBesu
         return { transactionReceipt: ledgerReceipt };
       default:
         throw new Error(
-          `${fnTag} Unrecognized ReceiptType: ${req.consistencyStrategy.receiptType}`,
+          `${fnTag} Unrecognized ReceiptType: ${request.consistencyStrategy.receiptType}`,
         );
     }
   }
 
+  public async transactPrivate(options: any): Promise<RunTransactionResponse> {
+    if (!this.web3EEA) {
+      await this.initializeEEAClient();
+    }
+
+    const txHash = await this.web3EEA?.eea.sendRawTransaction(options);
+    if (!txHash) {
+      throw new RuntimeError(`eea.sendRawTransaction provided no tx hash.`);
+    }
+    return this.getPrivateTxReceipt(options.privateFrom, txHash);
+  }
+
+  public async getPrivateTxReceipt(
+    privateFrom: string,
+    txHash: string,
+  ): Promise<RunTransactionResponse> {
+    const txPoolReceipt = await this.web3EEA?.priv.getTransactionReceipt(
```

Comment:
@travis-payne Once you refactor the if statement above to throw instead, the `?` will no longer be necessary because the compiler will understand that the type has been narrowed down sufficiently.